ci/osx-deps

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -euo pipefail
+
+fold_start() {
+    key=$1
+    title=$2
+    echo -e "travis_fold:start:$key\e[2K"
+    echo -e "travis_time:start:$key\e[2K"
+    tick="$(date +%s)"
+    echo "$title"
+}
+
+fold_end() {
+    key=$1
+    tock="$(date +%s)"
+    nano=000000000
+    echo -e "travis_time:end:$key:start=$tick$nano,finish=$tock$nano,duration=$((tock - tick))$nano\e[2K"
+    echo -e "travis_fold:end:$key\e[2K"
+}
+
+fold_start Python "Install Python 3.8 from python.org"
+curl -O https://www.python.org/ftp/python/3.8.5/python-3.8.5-macosx10.9.pkg
+sudo installer -package python-3.8.5-macosx10.9.pkg -target /
+sudo ln -s /usr/local/bin/python3 /usr/local/bin/python
+hash -r
+fold_end Python
+
+fold_start ccache 'Install ccache (compile it ourselves)'
+curl -O -L https://github.com/ccache/ccache/releases/download/v3.7.11/ccache-3.7.11.tar.xz
+tar xf ccache-3.7.11.tar.xz
+pushd ccache-3.7.11
+./configure --prefix=/usr/local
+make
+make install
+popd
+for compiler in clang clang++ cc gcc c++ g++; do
+  ln -sf ccache /usr/local/bin/$compiler
+done
+fold_end ccache
+
+fold_start pkg-config 'Install pkg-config (compile it ourselves)'
+curl -O -L https://dl.bintray.com/homebrew/mirror/pkg-config-0.29.2.tar.gz
+tar xf pkg-config-0.29.2.tar.gz
+pushd pkg-config-0.29.2
+./configure --disable-debug --prefix=/usr/local --disable-host-tool \
+            --with-internal-glib --with-pc-path=/usr/local/lib/pkgconfig:/usr/local/share/pkgconfig:/usr/local/pkgconfig \
+            --with-system-include-path=/usr/include
+make
+make install
+popd
+fold_end pkg-config
+
+fold_start libpng 'Install libpng (compile it ourselves)'
+curl -O -L https://downloads.sourceforge.net/libpng/libpng-1.6.37.tar.xz
+tar xf libpng-1.6.37.tar.xz
+pushd libpng-1.6.37
+./configure --disable-dependency-tracking --disable-silent-rules --prefix=/usr/local
+make
+make install
+popd
+fold_end libpng
+
+fold_start freetype 'Install freetype (just unpack into the build directory)'
+curl -O -L https://download.savannah.gnu.org/releases/freetype/freetype-2.6.1.tar.gz
```

Comment:
Why can't we just rely on what's already going on in `setupext.py`?

Reply:
I'm sure I added this because of some reason, but the details are lost in the rebases... I'll see if and how the build fails without this step.

Reply:
The [build](https://travis-ci.com/github/matplotlib/matplotlib/jobs/367595357) ends with the error message

> error: Failed to download FreeType. Please download one of ['https://downloads.sourceforge.net/project/freetype/freetype2/2.6.1/freetype-2.6.1.tar.gz', 'https://download.savannah.gnu.org/releases/freetype/freetype-2.6.1.tar.gz'] and extract it into build/freetype-2.6.1 at the top-level of the source repository.

I don't know how it is trying to download the file, but perhaps the older MacOS version doesn't support it?

Reply:
It's just using standard library `urllib`. Two possibilities, the SSL certificates are old/not found, or the cache location is not writable.